### ISSUE TITLE
Nested routes buggy with latest version of riot

### ISSUE THREAD

**Issue by danhper** [gabrielmoreira/riot-router#22]
Hi,

After upgrading Riot to the latest version, I am having issues with nested routes.
It seems that nested `route` tags are not mounted properly.
With the following structure (from the example)
- `/` (`<home />`)
- `/users` (`<users />`)
  - `/user/:userId` (`<user />`)
## One level nesting
1. When I go to `/users/user/123` and refresh the page, the `user` tag is not shown.
2. If I then click on `/users/user/456`, the `user` tag shows properly.
3. If I navigate to `/` and come back, the `user` tag displays properly.
## Two levels nesting

Now, I create a `<hello>` tag, add it as a default route under `/user/:userId` and add `<route />`
somewhere in the `<user />` tag.
1. When I go to `/users/user/123` and refresh the page, the `user` tag is not shown.
2. If I then click on `/users/user/456`, the `user` tag shows properly but `hello` tag does not show
3. If I then click on `/users/user/123`, again, both `user` tag and `hello` tags show properly
4. If I navigate to `/` and come back, `user` tag shows properly but `hello` tag does not.

To reproduce the one level nesting, make a fresh clone of the repository, run `npm install` and just try the steps in `example-01.html`.

To reproduce the two level nesting, I created a JSFiddle.
https://jsfiddle.net/47yzs5ea/

Any idea of what is going on here? I am trying to debug it but I am not sure what is happening.

A temporary bad workaround is to call `riot.router.start()` twice,
making the nested routes being processed on the second invocation of `Router.process`.
It probably fails with three levels of nesting but I did not try it.

**Comment by gabrielmoreira** [comment]
Hi @tuvistavie , sorry this delay.

I found three problems here.

1) By definition, you are using {updatable: true} on user route, but doesn't defined 
this.on('update', your update function to refresh content) like on users tag.

2) There is a bug, when a nested <route/> is mounted but not updated at first time. It's working when is next time, because route will listen to route:updated event.
I fixed this bug on riot-router <route/>, doing a updateRoute when tag is mounted.

3) There is a bug, that nested <route/> is leaking one old instance for this deep case. I will investigate this problem...
